consensus/polybft/bridge/bridge_manager.go

```diff
@@ -315,55 +322,63 @@ func (b *bridgeEventManager) AddLog(chainID *big.Int, eventLog *ethgo.Log) error
 		return err
 	}
 
-	if err := b.state.insertBridgeMessageEvent(event); err != nil {
+	if err := b.state.insertBridgeMessageEvent(event, nil); err != nil {
 		b.logger.Error("could not save bridge message event to boltDb", "err", err)
 
 		return err
 	}
 
-	if err := b.buildExternalBridgeBatch(nil); err != nil {
-		// we don't return an error here. If bridge message event is inserted in db,
-		// we will just try to build a batch on next block or next event arrival
-		b.logger.Error("could not build a batch on arrival of new bridge message event",
-			"err", err, "bridgeMessageID", event.ID)
-	}
-
 	return nil
 }
 
 // BridgeBatch returns a batch to be submitted if there is a pending batch with quorum
-func (b *bridgeEventManager) BridgeBatch(blockNumber uint64) (*BridgeBatchSigned, error) {
+func (b *bridgeEventManager) BridgeBatch(
```

Comment:
Why not changing this function to return a slice of `BridgeBatchSigned`, and remove this chain type parameter. You would have a local function called `getLargestPendingBatch` that would receive an appropriate pendingBridgeBatch list, and you would just call it twice in this function. Something like this:

```go
func (b *bridgeEventManager) BridgeBatch(blockNumber uint64) ([]*BridgeBatchSigned, error) {
     getLargestPendingBatchFn := func(pendingBatches []*PendingBridgeBatch) (*BridgeBatchSigned, error) {
            // logic
      }
      
      largestExternalBatch, err := getLargestPendingBatchFn(b.pendingBridgeBatchesExternal)
     if err != nil {
          return nil, fmt.Errorf("failed to get largest pending external batch: %w", err)
     }
     
     largestInternalBatch, err := getLargestPendingBatchFn(b.pendingBridgeBatchesInternal)
      if err != nil {
          return nil, fmt.Errorf("failed to get largest pending internal batch: %w", err)
     }
     
     var batches []*BridgeBatchSigned
     if largestExternalBatch != nil {
          batches = append(batches, largestExternalBatch)
     }
     
     if largestInternalBatch != nil {
          batches = append(batches, largestInternalBatch)
     }
     
     return batches, nil
}
```